### ISSUE TITLE
Latest preview assemblies are not signed

### ISSUE THREAD

**Issue by InKahootz** [CommunityToolkit/dotnet#254]
### Describe the bug

Since v8.0.0-preview.1, the toolkit DLL is no longer strong name signed and cannot be loaded by .NETFramework libraries/executables that are strong named.

### Regression

Yes. v7.1.2 is signed.

### Steps to reproduce

```text
1. Have a WPF app that targets both `net48` and `net6.0-windows` and is strong named.
2. Try to use the latest preview.
3. Works for NET 6.0 (because strong names no longer matter there) but not for net48.
```

### Expected behavior

CommunityToolkit.Mvvm.dll should be signed with the toolkit.snk that's in the root directory.

### Screenshots

v7.1.2  

![image](https://user-images.githubusercontent.com/1909698/168447621-f08b4a61-91fd-4f2d-91fa-873fba3f096e.png)

v8.0.0-preview

![image](https://user-images.githubusercontent.com/1909698/168447632-e4c0fdcf-6c2c-4703-aa71-a1b0df0db0eb.png)

### IDE and version

VS 2022 Preview

### IDE version

17.3-p1

### Nuget packages

- [ ] CommunityToolkit.Common
- [ ] CommunityToolkit.Diagnostics
- [ ] CommunityToolkit.HighPerformance
- [X] CommunityToolkit.Mvvm (aka MVVM Toolkit)

### Nuget package version(s)

v8.0.0-preview.3

### Help us help you

Yes, I'd like to be assigned to work on this item

**Comment by Sergio0694** [comment]
@michael-hawker would you know whether we're missing any steps in the CI to get the assemblies to be signed?
I though that was a requirement since we're publishing through the .NET Foundation, is the issue just for the preview ones?

**Comment by InKahootz** [comment]
In the original WindowCommunityTookit, [this](https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/cca764fafb2993311eaaadbab13be52506d656aa/Directory.Build.targets#L11) portion in the Directory.Build.targets handled it.

**Comment by Sergio0694** [comment]
@InKahootz I've assigned you to this work item 🙂
Please keep the check, we should only be signing assemblies for our core projects, not for the tests too.

**Comment by InKahootz** [comment]
@Sergio0694 Guidance on [CS1726](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1726?f1url=%3FappId%3Droslyn%26k%3Dk(CS1726))?

TL;DR Strong-named assemblies can only grant internal access (`InternalsVisibleToAttribute`) to other strongly signed assemblies.

Any downsides to signing the unit tests? 
I've always signed everything in my personal projects. Never had any issues.
I checked dotnet/roslyn and dotnet/runtime. Everything (managed) strong-named.

I'm going to go ahead and sign everything and put up a PR. We can discuss if there are alternatives.
